### ISSUE TITLE
Roberta: A robustly optimized bert pretraining approach

### ISSUE THREAD

**Issue by sooooner** [sooooner/papers#55]
# Roberta: A robustly optimized bert pretraining approach [[link](https://arxiv.org/abs/1907.11692)]

**Abstract**
BERT는 아직 배고프다.

## Introduction
BERT를 다른 방식으로 학습해봄
1. training the model longer, with bigger batches, over more data
2. removing the next sentence prediction objective
3. training on longer sequences
4. dynamically changing the masking pattern applied to the training data.

## Background
BERT 설명

###  Implementation
peak learning rate, number of warmup steps을 제외하곤 기존 BERT의 하이퍼 파라미터 따름  
훈련이 Adam의 epsilon에 매우 민감해서 경우에 따라 튜닝후 성능 올라감, 또 더큰 배치에서 안정성은 beta2=0.98로 설정하면 좋음  
대부분 seq의 길이 T는 512로 맞춤(max length)

### Data
10배 사용(160GB)

## Training Procedure Analysis
BERTBASE와 동일하게  (L = 12, H = 768, A = 12, 110M params)로 학습

### Static vs. Dynamic Masking
모든 Epoch에서 동일한 mask를 사용하지 않도록 데이터를 10번 복제하여 40 Epoch에 걸쳐 10가지 다른 방식으로 마스킹함   
즉, 같은 masking은 네번만 보도록 함 결과로 조금 더 좋아짐

### Model Input Format and Next Sentence Prediction
AlBERT를 보면 NSP가 중요한지 의문임

4가지 방법으로 실험해봄
1. SEGMENT-PAIR+NSP
BERT에서 사용한 방식
2. SENTENCE-PAIR+NSP
같은 문서에서 나온 인접 문장, batch size 키움
3. FULL-SENTENCES
연속된 문장으로 구성, NSP 제거
4. DOC-SENTENCES
다른 문단으로 넘어 가지 않음, NSP 제거

![image](https://user-images.githubusercontent.com/31437443/124102015-ecf17d00-da9a-11eb-95b5-7a1922137828.png)

결과 표를 볼때 NSP 불필요함

### Training with large batches
배치를 늘렸더니 perplexity좋아짐

### Text Encoding
50k vocab 사용함 데이터가 늘었으니 당연한건가?

##  Conclusion
더 많은 데이터, 더 큰 배치, 더 오래 훈련, NSP 제거, 더긴 seq, dynamic masking, 
 -> 성능 향상
